python/ray/data/_internal/execution/operators/map_operator_state.py

```diff
@@ -201,3 +218,62 @@ class _TaskState:
 
     inputs: RefBundle
     output: Optional[RefBundle] = None
+
+
+class _OutputQueue:
+    """Interface for swapping between different output order modes."""
+
+    def add_pending_task(self, task: "_TaskState"):
+        """Track the addition of a new task."""
+        pass
+
+    def notify_task_completed(self, task: "_TaskState"):
+        """Called when a previously added task completes."""
```

Comment:
Not quite clear to me about these two methods: the `_OrderedOutputQueue` impl below doesn't implement `add_pending_task`, so there is no "previously added task".